models/sessions.go

```diff
@@ -94,8 +94,8 @@ func FindSessionByUserID(tx *storage.Connection, userId uuid.UUID) (*Session, er
 	return session, nil
 }
 
-func InvalidateOtherFactorAssociatedSessions(tx *storage.Connection, currentSessionID, userID, factorID uuid.UUID) error {
-	return tx.RawQuery("DELETE FROM "+(&pop.Model{Value: Session{}}).TableName()+" WHERE user_id = ? AND factor_id = ? AND id != ?", userID, factorID, currentSessionID).Exec()
+func UpdateOtherFactorAssociatedSessions(tx *storage.Connection, currentSessionID, userID, factorID uuid.UUID, aal string) error {
+	return tx.RawQuery("UPDATE "+(&pop.Model{Value: Session{}}).TableName()+" set aal = ? WHERE user_id = ? AND factor_id = ? AND id != ?", aal, userID, factorID, currentSessionID).Exec()
```

Comment:
why do we not update the current session here? is this function meant to only update other sessions and not the existing one?

i see that we're also using `UpdateOtherFactorAssociatedSessions` in unenroll and we're still updating the `factor_id` after we destroy the factor - is this intended? seems like it would result in a non-existent foreign key error of some sort

would `tx.Destroy(factor)` result in a cascading delete? if so, it might also remove the session referencing it

Reply:
> would tx.Destroy(factor) result in a cascading delete? if so, it might also remove the session referencing it

hmm just checked, seems like there's no foreign key association between `sessions.factor_id` and `mfa_factors.id`, so there shouldn't be a cascading delete - but i'm not too sure why `sessions.factor_id` isn't referencing `mfa_factors.id` though

it might be cleaner to make this method a private method in the `models` package, and then use it in `models/factors.go` as a struct method 

```
func (f *Factor) DowngradeSession(tx *storage.Connection, sessionID string) {
  return updateOtherFactorAssociatedSessions(tx, sessionID, f.UserID, f.ID, "aal1")
}

# likewise for upgrading a session
...
```

Reply:
Good catch! The implementation is off. Loosely, the Initial intention was to account for the possibility that other factors might be present(e.g. another webauthn factor). However, this is based on another set of currently invalid assumptions.  Have ammended accordingly.

Re: `sessions.factor_id` and `mfa_factors.id` . They're not linked because of the cascading delete -- sessions can outlive the factors they are associated with. There's probably a cleaner way of implementing but haven't figured it out yet. I guess we can manually clear the `factorID` field when downgrading  --  will amend accordingly -- will also create a method in the models package as suggested above.